File: Kotlin/lab2task3/lab2task3.kt
import java.util.Scanner
import java.util.ArrayList

fun main() {
    // Создаем объект Scanner для ввода с консоли
    val input = Scanner(System.`in`)
    // Создаем список для хранения обратных чисел
    val nums = ArrayList<Int>()

    // Запрашиваем количество чисел у пользователя
    print("Введите количество чисел: ")
    // Считываем количество чисел
    val n = input.nextInt()

    // Цикл для ввода чисел и вычисления обратных чисел
    for (i in 0 until n) {
        // Запрашиваем у пользователя ввод числа
        print("Число $i: ")
        // Считываем введенное число
        var elem = input.nextInt()
        // Переменная для хранения обратного числа
        var revElem = 0

        // Цикл для вычисления обратного числа
        while (elem > 0) {
            revElem = revElem * 10 + elem % 10
            elem /= 10
        }

        // Добавляем обратное число в список
        nums.add(revElem)
    }

    // Выводим обратные числа
    for (elem in nums) {
        print("$elem ")
    }

}

File: C++/lab2task3/lab2task3.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    setlocale(LC_ALL, "Russian");
    vector<int> nums;

    int n;
    cout << "Введите количество символов: ";
    cin >> n;

    // Проверяем, что введенное количество символов неотрицательное число
    if (n <= 0) {
        cout << "Ошибка: введено некорректное количество символов." << endl;
        return 1; // Возвращаем ненулевой код, чтобы указать на ошибку
    }

    for (int i = 0; i < n; i++) {
        int elem;
        cout << "Элемент: ";
        cin >> elem;
        int revElem = 0;
        while (elem > 0) {
            revElem = revElem * 10 + elem % 10;
            elem = elem / 10;
        }
        nums.push_back(revElem);
    }

    for (auto elems : nums) {
        cout << elems << " ";
    }
    return 0;
}
